tests/lib/cpp/cxx/prj.conf

```diff
@@ -5,3 +5,9 @@ CONFIG_ZTEST_STACK_SIZE=2048
 CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE=128
 CONFIG_ZTEST_NEW_API=y
 CONFIG_CRC=y
+
+# RTIO
```

Comment:
There was a bug (linked in the PR) that RTIO wasn't working on C++ so this was added to avoid regression. I can't speak for other APIs but most likely the answer is that they should.